whylog/tests/tests_config/test_concated_regex.py

```diff
@@ -0,0 +1,240 @@
+        parser4 = RegexParserFactory.create_from_intent(parser_intent4)
+
+        content = "2015-12-03 12:11:00 Data is missing at alfa21. Loss = 567.02 GB. Host name: 101"
+
+        concated = ConcatedRegexParser([parser1, parser2, parser3, parser4])
+
+        assert concated.get_extracted_regex_params(content) == {
+            parser1.name: ["2015-12-03 12:11:00", "alfa21", "567.02", "101"],
+            parser2.name: ["2015-12-03 12:11:00", "alfa21. Loss = 567.02 GB. Host name: 101"],
+        }
+
+        concated = ConcatedRegexParser([parser4, parser3, parser2, parser1])
+
+        assert concated.get_extracted_regex_params(content) == {
+            parser1.name: ["2015-12-03 12:11:00", "alfa21", "567.02", "101"],
+            parser2.name: ["2015-12-03 12:11:00", "alfa21. Loss = 567.02 GB. Host name: 101"],
+        }
+
+    def test_single_subregex(self):
+        regex1 = "^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) Data is missing at (.*)\. Loss = (.*) GB\. Host name: (.*)$"
+        regex2 = "^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) Data is missing at (.*)$"
+
+        parser_intent1 = UserParserIntent("lostdata", "filesystem", regex1, [1], {1: to_date})
+        parser_intent2 = UserParserIntent("onlymissdata", "filesystem", regex2, [1], {1: to_date})
+
+        parser1 = RegexParserFactory.create_from_intent(parser_intent1)
+        parser2 = RegexParserFactory.create_from_intent(parser_intent2)
+
+        content = "2015-12-03 12:11:00 Data is missing at alfa21. Loss = 567.02 GB. Host name: 101"
+
+        concated = ConcatedRegexParser([parser1])
+
+        assert concated.get_extracted_regex_params(content) == {
+            parser1.name: ["2015-12-03 12:11:00", "alfa21", "567.02", "101"],
+        }
+
+        concated = ConcatedRegexParser([parser2])
+
+        assert concated.get_extracted_regex_params(content) == {
+            parser2.name: ["2015-12-03 12:11:00", "alfa21. Loss = 567.02 GB. Host name: 101"],
+        }
+
+    def test_large_concated_regex(self):
+        regex1 = "^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) Data is missing at (.*)\. Loss = (.*) GB\. Host name: (.*)$"
+        regex2 = "^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) Data is missing at (.*)$"
+        regex3 = "^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) Connection error occurred on (.*)\. Host name: (.*)$"
+        regex4 = "^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) Data migration from (.*) to (.*) failed\. Host name: (.*)$"
+
+        size = 100
+
+        parser_intent1 = UserParserIntent("lostdata", "filesystem", regex1, [1], {1: to_date})
+        parser_intent2 = UserParserIntent("onlymissdata", "filesystem", regex2, [1], {1: to_date})
+
+        base_list = []
+
+        for i in six.moves.range(size):
+            if i % 2 == 0:
+                base_list.append(UserParserIntent(uuid.uuid4(), "hydra", regex3, [1], {1: to_date}))
+            else:
+                base_list.append(UserParserIntent(uuid.uuid4(), "hydra", regex4, [1], {1: to_date}))
```

Comment:
clone of 2 lines above, refactor by setting variable in if statement and call `.append` just once